src/streams.h

```diff
@@ -142,7 +142,7 @@ class CDataStream
             vch.insert(it, first, last);
     }
 
-    void insert(iterator it, const char *first, const char *last)
+    void insert(iterator it, std::string::iterator first, std::string::iterator last)
     {
         if (last == first)
             return;
```

Comment:
since you are in here how about changing the assert below to DbgAssert(last - first > 0, return).  So in release mode, nothing is inserted if the range you offer is inverted.

Reply:
If we passed a negative value to memcopy won't we potentially corrupt the memory and cause a crash?  As for myself I don't often compile in debug mode because I work in a VM so I wouldn't even know I had a problem until much later, so I'd like to keep them as is if that's ok.

Reply:
well the DbgAssert would assert in debug mode and return in release mode so the memcpy would never be hit.  but its fine can keep the assert...